src/oop/object_interactions/medium/q3/MainTest.java

```diff
@@ -0,0 +1,225 @@
+                {2002, 10, 23, "Story"},
+                {1994, 2, 1, "War"},
+                {2017, 7, 23, "Victoy"}
+        };
+        for(int i = 0; i < tests.length; i ++) {
+            Object[][] arguments = {
+                    {tests[i][0], int.class},
+                    {tests[i][1], int.class},
+                    {tests[i][2], int.class},
+                    {tests[i][3], String.class}
+            };
+            Object testInstance = testClass.createInstance(arguments);
+            assertTrue(testClass.hasMethod(getMyYearMethodName, null, int.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyYearMethodName));
+            Object getMethodOutput = testClass.callMethod(getMyYearMethodName, testInstance);
+            _assertEquals(tests[i][0], getMethodOutput,
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyYearMethodName));
+
+            assertTrue(testClass.hasMethod(getMyMonthMethodName, null, int.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyMonthMethodName));
+            getMethodOutput = testClass.callMethod(getMyMonthMethodName, testInstance);
+            _assertEquals(tests[i][1], getMethodOutput,
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyMonthMethodName));
+
+            assertTrue(testClass.hasMethod(getMyDayMethodName, null, int.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyDayMethodName));
+            getMethodOutput = testClass.callMethod(getMyDayMethodName, testInstance);
+            _assertEquals(tests[i][2], getMethodOutput,
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyDayMethodName));
+
+            assertTrue(testClass.hasMethod(getMyStoryMethodName, null, String.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyStoryMethodName));
+            getMethodOutput = testClass.callMethod(getMyStoryMethodName, testInstance);
+            _assertEquals(tests[i][3], getMethodOutput,
+                    String.format("You have modified the provided %s method. Please revert it to the original state.", getMyStoryMethodName));
+        }
+
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(2020, 5, 17, "It's a new day!"),
+                Arguments.of(2017, 4, 3, "A new brand of car is out!"),
+                Arguments.of(2023, 4, 5, "The classes are cancelled!")
+        );
```

Comment:
You should also add a test case for all the ints being 0 and having an empty string.